mjattbot.js

```diff
@@ -58,33 +60,42 @@ function pokeCatch() {
 }
 
 client.on("message", message => {
-  if (message.author.bot) return;
-  if (message.channel.type === "dm") {
-    message.channel.send(
-      "Please try this in a public guild. Error: DM channel"
-    );
-    return;
-  }
-  if (!message.content.startsWith(key.prefix)) return;
+  if (message.author.bot) return; // ignore bots to avoid infinite loop
+  if (message.channel.type === "dm") return message.channel.send("Please try this in a public guild. Error: DM channel"); // cleanup, code looks bit cleaner here imo
+  
+  if (!message.content.startsWith(key.prefix)) return; // ignore msgs without prefix
 
   let command = message.content.split(" ")[0];
   command = command.slice(key.prefix.length);
+  // console.log(command);
 
   let args = message.content.split(" ").slice(1);
 
   try {
     let cmd = require(`./commands/${command}.js`);
-    cmd.run(client, message, args);
     count++;
     console.log(
-      `Message Number: ${count} Command: ${command} Args: ${args.join(" ")} Author: ${message.author.username}`
+      `Message Number: ${count} Command: ${command} Args: ${args.join(" ") ? args.join(" ") : "no arguments"} Author: ${message.author.username}`
```

Comment:
Might be better do do a length check rather than calling `args.join(" ")` twice. Something like this? `${args.length > 0 ? args.join(" ") : "no arguments"}`

Reply:
Yep, that is what I've also been thinking. Should do that instead of calling it twice